### PR TITLE
Add azure-cli-persistence copied from was-cli-persistence

### PR DESCRIPTION
**Pull request by robinmordasiewicz**
#58 the upstream repo does not seem to be maintained and this repo holds more comprehensive and maintained features. Please add azure-cli-persistence to this repo.

### CODE REVIEW

File: src/azure-cli-persistence/devcontainer-feature.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "Azure CLI Persistence",
+    "id": "azure-cli-persistence",
```
Comment (joshuanianji) [inline src/azure-cli-persistence/devcontainer-feature.json:3]: I think devcontainer feature IDs need to be unique - maybe change this to `azure-cli-persistence-new` or something similar? I think the name can stay the same.

We can also update the description field to something similar to "(fork of stuartleeks): <description>"

File: test/azure-cli-persistence/_default.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+set -e
+
+# This script is the "default" test script for the azure-cli-persistence feature.
+# It is not run as a scenario, but is run by other test scripts.
+
+# Optional: Import test library
+source dev-container-features-test-lib
+
+# check that `azure --help` works
+check "help" bash -c "azure help | grep 'usage'"
```
Comment (joshuanianji) [inline test/azure-cli-persistence/_default.sh:12]: Think tests are failing because this should be `az help, but other than that the tests seem good

**Comment by joshuanianji** [comment]
Hey @robinmordasiewicz, thanks for opening this PR! I'll be willing to maintain azure-cli-persistence in my repo.

Added a couple quick comments to fix `devcontainer-feature.json` and some tests, and after that I'm good to merge!

**Comment by joshuanianji** [comment]
Hey @robinmordasiewicz, just checking in. Do you still want to finish this PR?

**Comment by robinmordasiewicz** [comment]
Oh sorry. I had assumed it was already done. And yep. It’s up to you if you wish to accept the PR. On May 9, 2024, at 12:51 PM, Joshua Ji ***@***.***> wrote:﻿
Hey @robinmordasiewicz, just checking in. Do you still want to finish this PR?

**Comment by joshuanianji** [comment]
Thanks for letting me know! I'll fix failing CI and then merge

**Pull request merged**